supportcleaner.py

```diff
@@ -0,0 +1,189 @@
+#!/usr/bin/env python3
+
+import argparse
+import os
+import re
+import sys
+import zipfile
+
+from tempfile import TemporaryDirectory
+
+# All files in the defined directories and all subdirectories will be cleaned.
+# Setting this to '.' or '/' will cause the tool to clean all existing files in the zip.
+LOGDIRS = [
+    '.',
+]
+
+MAX_TMP_DIR_SIZE = (200 * 1024 * 1024)
+
+TMPDIR = TemporaryDirectory()
+
+if sys.version_info < (3, 5):
+    raise Exception('Python in version 3.5 or higher is required to run this tool.')
+
+
+def add_unit_prefix(num: float, suffix='B') -> str:
+    """
+    source: https://stackoverflow.com/questions/1094841/reusable-library-to-get-human-readable-version-of-file-size
+    """
+    for unit in ['', 'Ki', 'Mi', 'Gi', 'Ti', 'Pi', 'Ei', 'Zi']:
+        if abs(num) < 1024.0:
+            return "%3.1f%s%s" % (num, unit, suffix)
+        num /= 1024.0
+    return "%.1f%s%s" % (num, 'Yi', suffix)
+
+
+def remove_unit_prefix(numstr: str) -> (float, str):
+    num, prefix, unit = re.match(pattern=r'(\d+\.?\d*)([KMGTPEZY]i)?(.*)', string=numstr).groups()
+    num = float(num)
+    for i in ['', 'Ki', 'Mi', 'Gi', 'Ti', 'Pi', 'Ei', 'Zi', 'Yi']:
+        if prefix == i:
+            return num, unit
+        else:
+            num *= 1024
+
+
+def get_free_disk_space(path: str):
+    s = os.statvfs(path)
+    return s.f_frsize * s.f_bavail
+
+
+def _arguments() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        description='CLI tool to clean Atlassian support.zip from various data',
+    )
+    parser.add_argument(
+        'supportzip',
+        help='Path to support zip file to be cleaned',
+    )
+    parser.add_argument(
+        'baseurl',
+        help='Base-URL of the corresponding system',
+    )
+    return parser.parse_args()
+
+
+def _prepare():
+    try:
+        os.remove('cleaned.zip')
+        print('Removed existing cleaned.zip')
+    except FileNotFoundError:
+        pass
+
+
+def _extract_zip(supportzip: str):
+    global MAX_TMP_DIR_SIZE
+    zipf = zipfile.ZipFile(supportzip, 'r')
+    uncompressed_size = _get_uncompressed_size(zipf)
+    while uncompressed_size > MAX_TMP_DIR_SIZE:
+        print('\nWARNING: Decompressed size of {uncomp} exceeds allowed MAX_TMP_DIR_SIZE of {max_size}.'.format(
+            uncomp=add_unit_prefix(uncompressed_size),
+            max_size=add_unit_prefix(MAX_TMP_DIR_SIZE),
+        ))
+        answer = input(
+            'Free disk space: {free_space}\n\n'
+            'Change MAX_TMP_DIR_SIZE to:\n'
+            '(Enter value, prefixes are allowed (KiB, MiB, GiB, ...); a to abort)\n'
+            ''.format(free_space=add_unit_prefix(get_free_disk_space(TMPDIR.name)))
+        )
+
+        if answer == 'a':
+            zipf.close()
+            print('Aborted by user.')
+            exit()
+
+        try:
+            MAX_TMP_DIR_SIZE, _ = remove_unit_prefix(answer)
```

Comment:
Hier wird nicht zwischen bits und Bytes unterschieden, das kriegt der user aber wahrscheinlich nicht mit.
Da wäre vielleicht wenigstens ein Hinweis ganz praktisch, oder man rechnet einfach in Bytes um.

Reply:
Ich dachte irgendwie das wäre immer Bytes?!? Aber das sind Bits? Hmm ...

Reply:
Also, die file_size von dem zip sind Bytes, und MAX_TMP_DIR_SIZE dann hoffentlich auch, aber der userinput könnte ja auch was anderes sein...